guppylang/prelude/_internal/compiler.py

```diff
@@ -438,3 +438,37 @@ def build_array_set(
     )
     array, swapped_elem = iter(builder.add_op(op, array, idx, elem))
     return array, swapped_elem
+
+
+class RangeCompiler(CustomCallCompiler):
+    def compile(self, args: list[Wire]) -> list[Wire]:
+        from typing import cast
+
+        from guppylang.compiler.expr_compiler import python_value_to_hugr
+        from guppylang.definition.custom import CustomFunctionDef
+
+        # Find the builtins module. Brute-force search!
```

Comment:
This is the worst bit! Custom *Checkers* get access to the Func + full Context, *Compilers* only get the CompiledGlobals which is rather hard to use